### DISCUSSION TITLE
Unable to resolve dependency [Parameter #0 [ <required> ]

### DISCUSSION THREAD

**Discussion by ryatkins** [livewire/livewire#3521] [1. Help] [answered]
I've had this happen a few times and I'm not sure how to resolve it.  

Using the old style `@livewire()` to call a nested component works, but calling `<livewire:component-name :param1="$param1" />` does not and errors out with Unable to resolve dependency.

```
    <livewire:bom-insert  :item_no="$this->item_no" :nextSeqNum="$this->nextSeqNum" />

    @livewire('bom-insert', ['item_no' => $this->item_no, 'nextSeqNum' => $this->nextSeqNum])
```

```
    public function mount($item_no, $nextSeqNum)
    {


        $this->item_no = $item_no;
        $this->nextSeqNum = $nextSeqNum;

    }
```

**Comment by joshhanley** [comment]
@ryatkins that seems unusual that the `@livewire` would work but the tag syntax not, as Livewire's tag compiler compiles it into the `@livewire` syntax.

You could use xdebug and put a breakpoint in the tag compiler, and check that it is correctly compiling the `@livewire` syntax and see if it is different to your manual use of it or not.

I've linked the code that does it below.

Hope this helps!

https://github.com/livewire/livewire/blob/adaa7f9d822a781db01f921341b6d59c544c93da/src/LivewireTagCompiler.php#L77-L92

**Comment by ryatkins** [comment]
Is there a compiled set of files that can be cleared?

I’ve run into this many times in my code base and end up just using the
@livewire syntax when livewire:component fails.

**Reply by joshhanley** [reply]
@ryatkins nah, it just takes the tag string and converts it to `@livewire` on the fly in the method I linked above.

Did you get a chance to see whether it was generating the same `@livewire` output from the tag as you would manually type?

**Comment by ryatkins** [comment, marked as answer]
Hi @joshhanley - I just gave it a try again and it still wasn't getting to that code spot you mentioned.

I changed my bound parameter from `$item_no` to `$itemNo` and that fixed it.

Thanks for the help!

**Reply by joshhanley** [reply]
@ryatkins ah interesting, that suggest to me that the regex was probably failing then due to that parameter name.

Glad you got it working though!

**Comment by ryatkins** [comment]
Do you think it is a Livewire issue in the regex in that file or happening
in Laravel?

**Reply by joshhanley** [reply]
Livewire I think. I'm just trying to see if I can replicate it locally now

**Comment by joshhanley** [comment]
@ryatkins so it looks like Livewire is converting the underscore property to camel case.

This is what the tag compiler is outputting 

```blade
@livewire('sample', ['itemNo' => $item_no])
```

Instead of it being this, hence why it's not working for you

```blade
@livewire('sample', ['item_no' => $item_no])
```

**Reply by joshhanley** [reply]
@ryatkins have submitted a fix PR #3525

**Reply by ryatkins** [reply]
Outstanding @joshhanley !! 

I’ve hit this so many times because my database is in snake case. This will be great for the DX of Livewire.  Thanks!!

**Reply by joshhanley** [reply]
@ryatkins no worries 🙂  if you can comment that on the PR, that'd be great!
